docs/rancher/cloud-provider.md

```diff
@@ -47,40 +47,91 @@ When spinning up an RKE cluster using the Harvester node driver, you can perform
 1. Select `Harvester(Out-of-tree)` option.
 
     ![](/img/v1.2/rancher/rke-cloud-provider.png)
-  
+
 2. Install `Harvester Cloud Provider` from the Rancher marketplace.
 
     ![](/img/v1.2/rancher/install-harvester-cloud-provider.png)
 
-  
+
 ### Deploying to the RKE2 Cluster with Harvester Node Driver
 
 When spinning up an RKE2 cluster using the Harvester node driver, select the `Harvester` cloud provider. The node driver will then help deploy both the CSI driver and CCM automatically.
 
   ![](/img/v1.2/rancher/rke2-cloud-provider.png)
 
-### Deploying to the RKE2 custom cluster (experimental)
+### Deploying to the Custom RKE2 Cluster with Harvester Cloud Provider (Experimental)
 
 ![](/img/v1.2/rancher/custom.png)
-1. Use `generate_addon.sh` to generate a cloud-config and place it into the directory `/etc/kubernetes/cloud-config` on every custom node.
+1. Generate cloud config data using the script `generate_addon.sh`, and then place the data on every custom node (directory: `/etc/kubernetes/cloud-config`).
+
+    ```
+    curl -sfL https://raw.githubusercontent.com/harvester/cloud-provider-harvester/master/deploy/generate_addon.sh | bash -s <serviceaccount name> <namespace>
+    ```
+
+    The output will look as follows:
+
+    ```
+    ########## cloud config ############
+    apiVersion: v1
+    clusters:
+    - cluster:
+        certificate-authority-data: <CACERT>
+        server: https://HARVESTER-ENDPOINT/k8s/clusters/local
+      name: local
+    contexts:
+    - context:
+        cluster: local
+        namespace: default
+        user: harvester-cloud-provider-default-local
+      name: harvester-cloud-provider-default-local
+    current-context: harvester-cloud-provider-default-local
+    kind: Config
+    preferences: {}
+    users:
+    - name: harvester-cloud-provider-default-local
+      user:
+        token: <TOKEN>
 
-   ```
-   curl -sfL https://raw.githubusercontent.com/harvester/cloud-provider-harvester/master/deploy/generate_addon.sh | bash -s <serviceaccount name> <namespace> 
-   ```
+    ########## cloud-init user data ############
+    write_files:
+    - encoding: b64
+      content: <CONTENT>
+      owner: root:root
+      path: /etc/kubernetes/cloud-config
+      permissions: '0644'
+    ```
 
-  :::note
+    :::note
 
-  The `generate_addon.sh` script depends on `kubectl` and `jq` to operate the Harvester cluster.
+    The `generate_addon.sh` script depends on `kubectl` and `jq` to operate the Harvester cluster.
 
-  The script needs access to the `Harvester Cluster` kubeconfig to work. You can find the `kubeconfig` file from one of the Harvester management nodes in the `/etc/rancher/rke2/rke2.yaml` path.
+    The script needs access to the `Harvester Cluster` kubeconfig to work. You can find the `kubeconfig` file from one of the Harvester management nodes in the `/etc/rancher/rke2/rke2.yaml` path.
```

Comment:
We could add a kindly reminder that the kubeconfig file `/etc/rancher/rke2/rke2.yaml` on the management node looks like the following:

```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: <redacted>
    server: https://127.0.0.1:6443
  name: default
```

It's error-prone that if the user executes the `generate_addon.sh` on the node directly with the kubeconfig file's `server` field unmodified. The generated cloud-config contains `127.0.0.1` which is not going to work later on. It needs to be the Harvester cluster's VIP address.

Reply:
I added more content to this block. Please take a look again. Thank you.